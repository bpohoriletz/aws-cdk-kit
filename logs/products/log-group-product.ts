import * as logs from 'aws-cdk-lib/aws-logs';

export class LogGroupProduct implements LogGroupProduct {
  logGroupName?: string;
  retention?: logs.RetentionDays;
}

export interface ILogGroupBuilder {
  props: LogGroupProduct;
  setName?(name: string | undefined): ILogGroupBuilder;
  setRetention?(days: logs.RetentionDays | undefined): ILogGroupBuilder;
  getResult(): logs.LogGroupProps;
}

export interface ILogGroupBuilderConstructor {
  new (): ILogGroupBuilder;
}
