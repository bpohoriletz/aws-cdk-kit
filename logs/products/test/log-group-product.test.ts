import { LogGroupProduct } from '../log-group-product';

describe('new LogGroupProduct', () => {
  it('has correct defaults', () => {
    expect(new LogGroupProduct()).toMatchSnapshot();
  });
});
