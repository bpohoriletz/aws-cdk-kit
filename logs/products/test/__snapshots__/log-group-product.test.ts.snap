// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new LogGroupProduct has correct defaults 1`] = `LogGroupProduct {}`;
